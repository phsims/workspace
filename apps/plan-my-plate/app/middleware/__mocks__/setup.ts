export const getSpoonConfig = () => ({
  baseUrl: 'http://mock_base_url.com',
   imgUrl: 'http://mock_img_url.com',
  headers: {
    'Content-Type': 'application/json',
    'X-API-Key': 'http://my_mock_api_key.com',
  },
  env: 'development',
});
