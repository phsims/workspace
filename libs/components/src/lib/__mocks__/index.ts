export * from './nutrientsData'
export * from './recipesData'