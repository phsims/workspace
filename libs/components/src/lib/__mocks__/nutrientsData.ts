export const mockNutrients = [
  {
    name: 'Calories',
    value: 316.49,
  },
  {
    name: 'Fat',
    value: 12.09,
  },
];
