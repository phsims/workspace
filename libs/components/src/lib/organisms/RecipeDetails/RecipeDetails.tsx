import Image from 'next/image';
import { Grid, Typography, Box } from '@mui/material';
import AccessAlarmRoundedIcon from '@mui/icons-material/AccessAlarmRounded';
import RestaurantRoundedIcon from '@mui/icons-material/RestaurantRounded';
import ThumbUpRoundedIcon from '@mui/icons-material/ThumbUpRounded';

import RatingComponent from '../../atoms/Rating/RatingComponent';
import formatTime from '../../utils/formatTime';
import Nutrients from '../../atoms/Nutrients/Nutrients';
import InfoIcon from '../../atoms/InfoIcon/InfoIcon';
import FillIcon from '../../atoms/FillIcon/FillIcon';
import TagComponent from '../../atoms/TagComponent/TagComponent';
import { Nutrient } from '../../types/Nutrient';

export interface RecipeDetailsProps {
  recipeImg: string;
  title: string;
  servings: number;
  readyInMinutes: number;
  nutrients: Nutrient[];
  aggregateLikes: number;
  healthScore: number;
  dishTypes: string[];
  cuisines: string[];
  diets: string[];
}

export function RecipeDetails({
  recipeImg,
  title,
  servings,
  readyInMinutes,
  nutrients,
  aggregateLikes,
  healthScore,
  dishTypes,
  cuisines,
  diets,
}: RecipeDetailsProps) {
  return (
    <>
      <Typography variant="h1">{title}</Typography>
      <Grid container spacing={{ xs: 2, md: 3 }}>
        <Grid item lg={6} md={5} xs={12}>
          <Image
            src={recipeImg}
            alt={title}
            title={title}
            width={556}
            height={370}
            style={{
              width: '100%',
              height: 'auto',
              borderRadius: '2rem',
            }}
          />
        </Grid>
        <Grid
          item
          lg={6}
          md={7}
          xs={12}
          sx={{ display: 'flex', flexDirection: 'row', gap: 2 }}
        >
          <Grid
            item
            sm={6}
            xs={12}
            sx={{
              display: 'flex',
              alignItems: 'stretch',
              flexDirection: 'column',
              justifyContent: 'space-between',
            }}
          >
            <Grid item xs={12}>
              <RatingComponent itemRating={0} />
              <InfoIcon
                icon={
                  <RestaurantRoundedIcon fontSize="medium" color="primary" />
                }
                infoTitle="Serves:"
                text={servings.toString()}
              />

              <InfoIcon
                icon={
                  <AccessAlarmRoundedIcon color="primary" fontSize="medium" />
                }
                infoTitle="Ready in:"
                text={formatTime(readyInMinutes)}
              />
            </Grid>

            <Grid
              item
              xs={12}
              sx={{
                display: 'flex',
                flexDirection: 'column',
                justifyContent: 'center',
              }}
            >
              <InfoIcon
                infoTitle={`Health score ${healthScore}`}
                icon={
                  <FillIcon
                    fillPercentage={healthScore}
                    ariaLabel={`Health score ${healthScore}`}
                  />
                }
              />
              {aggregateLikes && (
                <InfoIcon
                  infoTitle={`${aggregateLikes} likes`}
                  icon={
                    <ThumbUpRoundedIcon color="secondary" fontSize="medium" />
                  }
                />
              )}
            </Grid>

            {nutrients?.length > 0 && (
              <Grid item xs={12}>
                <Box>
                  <Nutrients
                    title="Nutritional information:"
                    nutrients={nutrients}
                  />
                </Box>
              </Grid>
            )}
          </Grid>
          <Grid
            item
            sm={6}
            xs={12}
            sx={{ display: 'flex', flexDirection: 'column', gap: 1 }}
          >
            {dishTypes?.length > 0 && (
              <TagComponent
                tags={dishTypes}
                varient="outlined"
                title="Great for:"
              />
            )}
            {cuisines?.length > 0 && (
              <TagComponent
                tags={cuisines}
                varient="outlined"
                color="success"
                title="Cuisine:"
              />
            )}
            {diets?.length > 0 && (
              <TagComponent
                tags={diets}
                varient="outlined"
                color="secondary"
                title="Suitable for:"
              />
            )}
          </Grid>
        </Grid>
      </Grid>
    </>
  );
}

export default RecipeDetails;
