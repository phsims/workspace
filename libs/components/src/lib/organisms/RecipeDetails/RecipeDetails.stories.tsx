import type { Meta, StoryObj } from '@storybook/react';
import { RecipeDetails } from './RecipeDetails';
import { Box } from '@mui/material';
import { within } from '@storybook/testing-library';
import { expect } from '@storybook/jest';
import { mockNutrients } from '../../__mocks__/nutrientsData';
import { mockRecipes } from '../../__mocks__/recipesData';


const meta: Meta<typeof RecipeDetails> = {
  component: RecipeDetails,
  title: '@organisms/RecipeDetails',
  decorators: [
    (Story) => (
      <Box sx={{ width: '900px', m: 3, background: 'grey.light' }}>
        <Story />
      </Box>
    ),
  ],
};
export default meta;
type Story = StoryObj<typeof RecipeDetails>;

export const Primary = {
  args: {
    ...mockRecipes.recipes[0],
    nutrients:mockNutrients,
    recipeImg: '/652825-556x370.webp',
  },
};

export const Heading: Story = {
  args: { ...Primary.args },
  play: async ({ canvasElement }) => {
    const canvas = within(canvasElement);
    expect(canvas.getByText(/My Asian Calamari/gi)).toBeInTheDocument();
  },
};
