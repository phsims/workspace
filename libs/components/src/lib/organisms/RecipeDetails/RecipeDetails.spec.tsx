import { render } from '@testing-library/react';

import RecipeDetails from './RecipeDetails';
import { mockNutrients } from '../../__mocks__/nutrientsData';
import { mockRecipes } from '../../__mocks__/recipesData';

const mockRecipe = {
  ...mockRecipes.recipes[0],
  nutrients: mockNutrients,
  recipeImg: '/652825-556x370.webp',
};
describe('RecipeDetails', () => {
  it('should render successfully', () => {
    const { baseElement } = render(<RecipeDetails {...mockRecipe} />);
    expect(baseElement).toBeTruthy();
  });

  it('should match snapshot', () => {
    const { baseElement } = render(<RecipeDetails {...mockRecipe} />);
    expect(baseElement).toMatchSnapshot();
  } );

  
});
