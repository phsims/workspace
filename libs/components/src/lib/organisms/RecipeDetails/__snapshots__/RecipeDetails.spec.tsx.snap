// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeDetails should match snapshot 1`] = `
<body>
  <div>
    <h1
      class="MuiTypography-root MuiTypography-h1 css-o2w69a-MuiTypography-root"
    >
      My Asian Calamari
    </h1>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-spacing-md-3 css-1xgjpwf-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-5 MuiGrid-grid-lg-6 css-tdvl3x-MuiGrid-root"
      >
        <img
          alt="My Asian Calamari"
          data-nimg="1"
          decoding="async"
          height="370"
          loading="lazy"
          src="/_next/image?url=%2F652825-556x370.webp&w=1200&q=75"
          srcset="/_next/image?url=%2F652825-556x370.webp&w=640&q=75 1x, /_next/image?url=%2F652825-556x370.webp&w=1200&q=75 2x"
          style="color: transparent; width: 100%; height: auto; border-radius: 2rem;"
          title="My Asian Calamari"
          width="556"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-7 MuiGrid-grid-lg-6 css-1c2y3me-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 css-rk5c46-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
          >
            <span
              class="MuiRating-root MuiRating-sizeLarge css-1usiq4q-MuiRating-root"
              color="warning.main"
            >
              <span
                class="MuiRating-decimal css-34he1w-MuiRating-decimal"
              >
                <label
                  class="css-dqr9h-MuiRating-label"
                  for=":rc:"
                  style="width: 0%; overflow: hidden; position: absolute;"
                >
                  <span
                    class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                      data-testid="StarBorderIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="m22 9.24-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiRating-visuallyHidden"
                  >
                    0.5 Stars
                  </span>
                </label>
                <input
                  class="MuiRating-visuallyHidden"
                  id=":rc:"
                  name="rating"
                  type="radio"
                  value="0.5"
                />
                <label
                  class="css-dqr9h-MuiRating-label"
                  for=":rd:"
                >
                  <span
                    class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                      data-testid="StarBorderIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="m22 9.24-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiRating-visuallyHidden"
                  >
                    1 Star
                  </span>
                </label>
                <input
                  class="MuiRating-visuallyHidden"
                  id=":rd:"
                  name="rating"
                  type="radio"
                  value="1"
                />
              </span>
              <span
                class="MuiRating-decimal css-34he1w-MuiRating-decimal"
              >
                <label
                  class="css-dqr9h-MuiRating-label"
                  for=":re:"
                  style="width: 0%; overflow: hidden; position: absolute;"
                >
                  <span
                    class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                      data-testid="StarBorderIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="m22 9.24-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiRating-visuallyHidden"
                  >
                    1.5 Stars
                  </span>
                </label>
                <input
                  class="MuiRating-visuallyHidden"
                  id=":re:"
                  name="rating"
                  type="radio"
                  value="1.5"
                />
                <label
                  class="css-dqr9h-MuiRating-label"
                  for=":rf:"
                >
                  <span
                    class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                      data-testid="StarBorderIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="m22 9.24-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiRating-visuallyHidden"
                  >
                    2 Stars
                  </span>
                </label>
                <input
                  class="MuiRating-visuallyHidden"
                  id=":rf:"
                  name="rating"
                  type="radio"
                  value="2"
                />
              </span>
              <span
                class="MuiRating-decimal css-34he1w-MuiRating-decimal"
              >
                <label
                  class="css-dqr9h-MuiRating-label"
                  for=":rg:"
                  style="width: 0%; overflow: hidden; position: absolute;"
                >
                  <span
                    class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                      data-testid="StarBorderIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="m22 9.24-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiRating-visuallyHidden"
                  >
                    2.5 Stars
                  </span>
                </label>
                <input
                  class="MuiRating-visuallyHidden"
                  id=":rg:"
                  name="rating"
                  type="radio"
                  value="2.5"
                />
                <label
                  class="css-dqr9h-MuiRating-label"
                  for=":rh:"
                >
                  <span
                    class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                      data-testid="StarBorderIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="m22 9.24-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiRating-visuallyHidden"
                  >
                    3 Stars
                  </span>
                </label>
                <input
                  class="MuiRating-visuallyHidden"
                  id=":rh:"
                  name="rating"
                  type="radio"
                  value="3"
                />
              </span>
              <span
                class="MuiRating-decimal css-34he1w-MuiRating-decimal"
              >
                <label
                  class="css-dqr9h-MuiRating-label"
                  for=":ri:"
                  style="width: 0%; overflow: hidden; position: absolute;"
                >
                  <span
                    class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                      data-testid="StarBorderIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="m22 9.24-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiRating-visuallyHidden"
                  >
                    3.5 Stars
                  </span>
                </label>
                <input
                  class="MuiRating-visuallyHidden"
                  id=":ri:"
                  name="rating"
                  type="radio"
                  value="3.5"
                />
                <label
                  class="css-dqr9h-MuiRating-label"
                  for=":rj:"
                >
                  <span
                    class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                      data-testid="StarBorderIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="m22 9.24-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiRating-visuallyHidden"
                  >
                    4 Stars
                  </span>
                </label>
                <input
                  class="MuiRating-visuallyHidden"
                  id=":rj:"
                  name="rating"
                  type="radio"
                  value="4"
                />
              </span>
              <span
                class="MuiRating-decimal css-34he1w-MuiRating-decimal"
              >
                <label
                  class="css-dqr9h-MuiRating-label"
                  for=":rk:"
                  style="width: 0%; overflow: hidden; position: absolute;"
                >
                  <span
                    class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                      data-testid="StarBorderIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="m22 9.24-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiRating-visuallyHidden"
                  >
                    4.5 Stars
                  </span>
                </label>
                <input
                  class="MuiRating-visuallyHidden"
                  id=":rk:"
                  name="rating"
                  type="radio"
                  value="4.5"
                />
                <label
                  class="css-dqr9h-MuiRating-label"
                  for=":rl:"
                >
                  <span
                    class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                      data-testid="StarBorderIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="m22 9.24-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiRating-visuallyHidden"
                  >
                    5 Stars
                  </span>
                </label>
                <input
                  class="MuiRating-visuallyHidden"
                  id=":rl:"
                  name="rating"
                  type="radio"
                  value="5"
                />
              </span>
              <label
                class="MuiRating-label MuiRating-pristine css-dqr9h-MuiRating-label"
              >
                <input
                  class="MuiRating-visuallyHidden"
                  id="rating-empty"
                  name="rating"
                  type="radio"
                  value=""
                />
                <span
                  class="MuiRating-visuallyHidden"
                >
                  Empty
                </span>
              </label>
            </span>
            <div
              class="MuiBox-root css-1f0wxn3"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
                data-testid="RestaurantRoundedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16 6v6c0 1.1.9 2 2 2h1v7c0 .55.45 1 1 1s1-.45 1-1V3.13c0-.65-.61-1.13-1.24-.98C17.6 2.68 16 4.51 16 6m-5 3H9V3c0-.55-.45-1-1-1s-1 .45-1 1v6H5V3c0-.55-.45-1-1-1s-1 .45-1 1v6c0 2.21 1.79 4 4 4v8c0 .55.45 1 1 1s1-.45 1-1v-8c2.21 0 4-1.79 4-4V3c0-.55-.45-1-1-1s-1 .45-1 1z"
                />
              </svg>
              <span
                class="MuiTypography-root MuiTypography-body1_Bold css-yylahi-MuiTypography-root"
              >
                Serves:
              </span>
              <p
                class="MuiTypography-root MuiTypography-body1 css-4z0fiv-MuiTypography-root"
              >
                4
              </p>
            </div>
            <div
              class="MuiBox-root css-1f0wxn3"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
                data-testid="AccessAlarmRoundedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="m15.87 15.25-3.37-2V8.72c0-.4-.32-.72-.72-.72h-.06c-.4 0-.72.32-.72.72v4.72c0 .35.18.68.49.86l3.65 2.19c.34.2.78.1.98-.24.21-.35.1-.8-.25-1m5.31-10.24L18.1 2.45c-.42-.35-1.05-.3-1.41.13-.35.42-.29 1.05.13 1.41l3.07 2.56c.42.35 1.05.3 1.41-.13.36-.42.3-1.05-.12-1.41M4.1 6.55l3.07-2.56c.43-.36.49-.99.13-1.41-.35-.43-.98-.48-1.4-.13L2.82 5.01c-.42.36-.48.99-.12 1.41.35.43.98.48 1.4.13M12 4c-4.97 0-9 4.03-9 9s4.03 9 9 9 9-4.03 9-9-4.03-9-9-9m0 16c-3.86 0-7-3.14-7-7s3.14-7 7-7 7 3.14 7 7-3.14 7-7 7"
                />
              </svg>
              <span
                class="MuiTypography-root MuiTypography-body1_Bold css-yylahi-MuiTypography-root"
              >
                Ready in:
              </span>
              <p
                class="MuiTypography-root MuiTypography-body1 css-4z0fiv-MuiTypography-root"
              >
                45 Minutes
              </p>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-15q8tmm-MuiGrid-root"
          >
            <div
              class="MuiBox-root css-1f0wxn3"
            >
              <div
                class="MuiBox-root css-djwqx3"
              >
                <svg
                  aria-hidden="true"
                  aria-label="Health score 12"
                  class="MuiSvgIcon-root MuiSvgIcon-colorSuccess MuiSvgIcon-fontSizeMedium css-4b67si-MuiSvgIcon-root"
                  data-testid="FavoriteBorderRoundedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19.66 3.99c-2.64-1.8-5.9-.96-7.66 1.1-1.76-2.06-5.02-2.91-7.66-1.1-1.4.96-2.28 2.58-2.34 4.29-.14 3.88 3.3 6.99 8.55 11.76l.1.09c.76.69 1.93.69 2.69-.01l.11-.1c5.25-4.76 8.68-7.87 8.55-11.75-.06-1.7-.94-3.32-2.34-4.28M12.1 18.55l-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05"
                  />
                </svg>
                <div
                  class="MuiBox-root css-v0jgcu"
                >
                  <svg
                    height="24"
                    viewBox="0 0 24 24"
                    width="24"
                  >
                    <defs>
                      <lineargradient
                        id="dynamic-gradient"
                        x1="0"
                        x2="0"
                        y1="100%"
                        y2="0"
                      >
                        <stop
                          offset="12%"
                          stop-color="#2e7d32"
                        />
                        <stop
                          offset="12%"
                          stop-color="transparent"
                        />
                      </lineargradient>
                    </defs>
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="FavoriteRoundedIcon"
                      focusable="false"
                      style="fill: url(#dynamic-gradient);"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M13.35 20.13c-.76.69-1.93.69-2.69-.01l-.11-.1C5.3 15.27 1.87 12.16 2 8.28c.06-1.7.93-3.33 2.34-4.29 2.64-1.8 5.9-.96 7.66 1.1 1.76-2.06 5.02-2.91 7.66-1.1 1.41.96 2.28 2.59 2.34 4.29.14 3.88-3.3 6.99-8.55 11.76z"
                      />
                    </svg>
                  </svg>
                </div>
              </div>
              <span
                class="MuiTypography-root MuiTypography-body1_Bold css-yylahi-MuiTypography-root"
              >
                Health score 12
              </span>
            </div>
            <div
              class="MuiBox-root css-1f0wxn3"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-colorSecondary MuiSvgIcon-fontSizeMedium css-kw0y8f-MuiSvgIcon-root"
                data-testid="ThumbUpRoundedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M13.12 2.06 7.58 7.6c-.37.37-.58.88-.58 1.41V19c0 1.1.9 2 2 2h9c.8 0 1.52-.48 1.84-1.21l3.26-7.61C23.94 10.2 22.49 8 20.34 8h-5.65l.95-4.58c.1-.5-.05-1.01-.41-1.37-.59-.58-1.53-.58-2.11.01M3 21c1.1 0 2-.9 2-2v-8c0-1.1-.9-2-2-2s-2 .9-2 2v8c0 1.1.9 2 2 2"
                />
              </svg>
              <span
                class="MuiTypography-root MuiTypography-body1_Bold css-yylahi-MuiTypography-root"
              >
                3 likes
              </span>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
          >
            <div
              class="MuiBox-root css-0"
            >
              <span
                class="MuiTypography-root MuiTypography-body1_Bold css-yylahi-MuiTypography-root"
              >
                Nutritional information:
              </span>
              <div
                class="MuiBox-root css-pg4npi"
              >
                <div
                  class="MuiBox-root css-1r6wdtb"
                >
                  <span
                    class="MuiTypography-root MuiTypography-body1_Bold css-1el569k-MuiTypography-root"
                  >
                    Calories
                  </span>
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-1k4wxgx-MuiTypography-root"
                  >
                    316.49
                  </p>
                </div>
                <div
                  class="MuiBox-root css-1r6wdtb"
                >
                  <span
                    class="MuiTypography-root MuiTypography-body1_Bold css-1el569k-MuiTypography-root"
                  >
                    Fat
                  </span>
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-1k4wxgx-MuiTypography-root"
                  >
                    12.09
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 css-1t4ylev-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-0"
          >
            <h6
              class="MuiTypography-root MuiTypography-h6 css-1kx63bm-MuiTypography-root"
            >
              Great for:
            </h6>
            <div
              class="MuiStack-root css-1dtqmwz-MuiStack-root"
            >
              <div
                class="MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorPrimary MuiChip-outlinedPrimary css-gww9aq-MuiChip-root"
              >
                <span
                  class="MuiChip-label MuiChip-labelMedium css-1jzq0dw-MuiChip-label"
                />
              </div>
            </div>
          </div>
          <div
            class="MuiBox-root css-0"
          >
            <h6
              class="MuiTypography-root MuiTypography-h6 css-1kx63bm-MuiTypography-root"
            >
              Cuisine:
            </h6>
            <div
              class="MuiStack-root css-1dtqmwz-MuiStack-root"
            >
              <div
                class="MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorSuccess MuiChip-outlinedSuccess css-rmbuyw-MuiChip-root"
              >
                <span
                  class="MuiChip-label MuiChip-labelMedium css-1jzq0dw-MuiChip-label"
                />
              </div>
            </div>
          </div>
          <div
            class="MuiBox-root css-0"
          >
            <h6
              class="MuiTypography-root MuiTypography-h6 css-1kx63bm-MuiTypography-root"
            >
              Suitable for:
            </h6>
            <div
              class="MuiStack-root css-1dtqmwz-MuiStack-root"
            >
              <div
                class="MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorSecondary MuiChip-outlinedSecondary css-3iing3-MuiChip-root"
              >
                <span
                  class="MuiChip-label MuiChip-labelMedium css-1jzq0dw-MuiChip-label"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
